### PR TITLE
15742 Fixed missing session storage variable and config dependency

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#15742

*Description of changes:*
- app version = 5.11.5a
- updated browsers list
- load configuration before checking business id
- set missing session storage variable
- check for empty URL before navigate

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
 {
   "name": "business-filings-ui",
-  "version": "5.11.5",
+  "version": "5.11.5a",
```
Comment (severinbeauvais) [inline package-lock.json:3]: This PR will be merged into the build that was in Test earlier today. (Main branch contains other fixes that we don't want to include.)

File: package-lock.json [hunk 2]
```diff
@@ -6569,9 +6569,9 @@
       }
     },
     "node_modules/caniuse-lite": {
-      "version": "1.0.30001410",
-      "resolved": "https://registry.npmjs.org/caniuse-lite/-/caniuse-lite-1.0.30001410.tgz",
-      "integrity": "sha512-QoblBnuE+rG0lc3Ur9ltP5q47lbguipa/ncNMyyGuqPk44FxbScWAeEO+k5fSQ8WekdAK4mWqNs1rADDAiN5xQ==",
+      "version": "1.0.30001469",
```
Comment (severinbeauvais) [inline package-lock.json:6572]: Updating browsers list needs to be done periodically.

File: src/utils/navigate.ts [hunk 3]
```diff
@@ -4,6 +4,8 @@
  */
 export function navigate (url: string): boolean {
   try {
+    if (!url) throw new Error('empty URL')
```
Comment (severinbeauvais) [inline src/utils/navigate.ts:7]: This will prevent the looping navigation (line 19 won't be run).

File: src/store/Configuration/mutations.ts [hunk 4]
```diff
@@ -21,6 +21,7 @@ export default {
   setSessionVariables (_state: ConfigurationStateIF, responseData: any) {
     // The following four session variables are used by SBC Header (a common component):
     sessionStorage.setItem('AUTH_WEB_URL', responseData['AUTH_WEB_URL'])
+    sessionStorage.setItem('BUSINESSES_URL', responseData['BUSINESSES_URL'])
```
Comment (severinbeauvais) [inline src/store/Configuration/mutations.ts:24]: Needed by navigate().

File: src/main.ts [hunk 5]
```diff
@@ -28,18 +28,19 @@ Vue.use(Vuelidate)
...
   const windowLocationPathname = window.location.pathname // eg, /business/CP1234567/...
   const windowLocationOrigin = window.location.origin // eg, http://localhost:8080
   const applicationUrl = windowLocationOrigin + processEnvBaseUrl
-  setBaseRouteAndBusinessId(windowLocationPathname, processEnvBaseUrl, windowLocationOrigin) // may throw an error
 
-  // fetch the store first as it has no dependencies
-  const store = getVuexStore()
+  // first load the configuration, then set base route and check business id
   await store.dispatch('loadConfiguration', applicationUrl)
+  setBaseRouteAndBusinessId(windowLocationPathname, processEnvBaseUrl, windowLocationOrigin) // may throw an error
```
Comment (severinbeauvais) [inline src/main.ts:43]: setBaseRouteAndBusinessId() may throw an error, which we cannot recover from if we haven't loaded the configuration first.

**Review by JazzarKarim — APPROVED** [review]
LGTM.

**Pull request merged**